cbor.js

```diff
@@ -23,9 +23,63 @@
  */
 
 (function(global, undefined) { "use strict";
-var POW_2_24 = 5.960464477539063e-8,
-    POW_2_32 = 4294967296,
-    POW_2_53 = 9007199254740992;
+var POW_2_32 = 4294967296,
+    POW_2_53 = 9007199254740992,
+    singleFloatView = new Float32Array(1),
+    singleIntView = new Uint32Array(singleFloatView.buffer);
+
+
+function getFloat16(value) {
+  singleFloatView[0] = value;
+  var fbits = singleIntView[0];
+  var sign = ( fbits >> 16 ) & 0x8000;
+  var exponentAndMantissa = fbits & 0x7fffffff;
+  var val = ( exponentAndMantissa ) + 0x1000;
+  if (val >= 0x47800000) {
+    /* jshint laxbreak:true */
+    return val < 0x7f800000
+      ? sign | 0x7c00
+      : sign | 0x7c00 | ( fbits & 0x007fffff ) >> 13;
+    /* jshint laxbreak:false */
+  }
+  if (val >= 0x38800000) {
+    return sign | val - 0x38000000 >> 13;
+  }
+  if (val < 0x33000000) {
+    return sign;
+  }
+  val = ( exponentAndMantissa ) >> 23;
+  return sign | ( ( fbits & 0x7fffff | 0x800000 ) +
+    ( 0x800000 >>> val - 102 )
+    >> 126 - val );
+}
```

Comment:
this varies slightly from known good encoding functions of half precision floats; one branch that depended on an mathematically impossible case was removed